addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/handler/LoxoneMiniserverHandler.java

```diff
@@ -230,12 +256,12 @@ public void onNewConfig(LxServer server) {
         logger.trace("Building new channels ({} controls)", server.getControls().size());
         for (LxControl control : server.getControls().values()) {
             List<Channel> newChannels = createChannelsForControl(control);
-            if (newChannels != null) {
-                channels.addAll(newChannels);
-                for (Channel channel : newChannels) {
-                    ChannelUID id = channel.getUID();
-                    controls.put(id, control);
-                }
+            channels.addAll(newChannels);
+            for (Channel channel : newChannels) {
+                @SuppressWarnings("null")
```

Comment:
Channel class seems to be null by default (it has some of the fields annotated with NonNull). But uid can be null and there actually is a (package protected only) default constructor where uid is not initialized. I thing this can be fixed in the Channel class, probably by simply initializing uid to a new ChannelUUID() in the default constructor and annotating this as non null. I will remove this suppression and leave the warning, then try fix it in the framework.